lib/samson/build_finder.rb

```diff
@@ -91,7 +91,8 @@ def possible_builds
       commits = [@job.commit]
 
       if defined?(SamsonKubernetes) && @job.deploy.kubernetes_reuse_build
-        previous = @job.deploy.previous_deploy&.job&.commit
+        previous_scope = @job.deploy.stage.deploys.prior_to(@job.deploy).where(kubernetes_reuse_build: false)
```

Comment:
do you need a sort by or is the implict order guaranteed?

Reply:
deploys are sorted `id desc` so the first will be the newest